upb/json/parser.rl

```diff
@@ -1345,7 +1574,10 @@ static void end_object(upb_json_parser *p) {
     value
     <: any >{ fhold; fret; } ;
 
-  main := ws object ws;
+  main :=
+    ws
+    (object | number_object | true_object | false_object | string_object)
```

Comment:
Why do you need to change "main"?  When do we need numbers or true/false/string at the top level?

To me number_object/true_object/false_object/etc. seem redundant since these cases are already handled in "value".

Reply:
For top level wrapper values, there is no  wrapping ```{}```. Currently, the definition of ```object``` starts with ```{``` and ends with ```}```.  So it cannot  accept top level wrapper values.  I also have test in protobuf to verify this.